src/app/fdctl/run/tiles/fd_plugin.c

```diff
@@ -0,0 +1,202 @@
+#include "../../../../disco/tiles.h"
+
+#include "generated/plugin_seccomp.h"
+
+#include "../../../../disco/plugin/fd_plugin.h"
+#include "../../../../flamenco/types/fd_types.h"
+
+typedef struct {
+  fd_wksp_t * mem;
+  ulong       chunk0;
+  ulong       wmark;
+  ulong       mtu;
+} fd_plugin_in_ctx_t;
+
+typedef struct {
+  fd_plugin_in_ctx_t in[ 64UL ];
+
+  fd_wksp_t * out_mem;
+  ulong       out_chunk0;
+  ulong       out_wmark;
+  ulong       out_chunk;
+} fd_plugin_ctx_t;
+
+FD_FN_CONST static inline ulong
+scratch_align( void ) {
+  return 128UL;
+}
+
+FD_FN_PURE static inline ulong
+scratch_footprint( fd_topo_tile_t const * tile ) {
+  (void)tile;
+  ulong l = FD_LAYOUT_INIT;
+  l = FD_LAYOUT_APPEND( l, alignof( fd_plugin_ctx_t ), sizeof( fd_plugin_ctx_t ) );
+  return FD_LAYOUT_FINI( l, scratch_align() );
+}
+
+FD_FN_CONST static inline void *
+mux_ctx( void * scratch ) {
+  return (void*)fd_ulong_align_up( (ulong)scratch, alignof( fd_plugin_ctx_t ) );
+}
+
+static inline void
+during_frag( void * _ctx,
+             ulong  in_idx,
+             ulong  seq,
+             ulong  sig,
+             ulong  chunk,
+             ulong  sz,
+             int *  opt_filter ) {
+  (void)seq;
+  (void)sig;
+  (void)opt_filter;
+
+  fd_plugin_ctx_t * ctx = (fd_plugin_ctx_t *)_ctx;
+
+  uchar * src = (uchar *)fd_chunk_to_laddr( ctx->in[ in_idx ].mem, chunk );
+  ulong * dst = (ulong *)fd_chunk_to_laddr( ctx->out_mem, ctx->out_chunk );
+
+  ulong _sz = sz;
+  if( FD_UNLIKELY( in_idx==1UL ) ) _sz = 8UL + 40200UL*(58UL+12UL*34UL);
+  else if( FD_UNLIKELY( in_idx==2UL ) ) _sz = 40UL + 40200UL*40UL; /* ... todo... sigh, sz is not correct since it's too big */
+
+  if( FD_UNLIKELY( chunk<ctx->in[ in_idx ].chunk0 || chunk>ctx->in[ in_idx ].wmark || _sz>ctx->in[ in_idx ].mtu ) )
+    FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz, ctx->in[ in_idx ].chunk0, ctx->in[ in_idx ].wmark ));
+
+  fd_memcpy( dst, src, _sz );
```

Comment:
Same comment as above. This is a huge memcpy that's going to create a decent amount of cache traffic, and it seems a bit unnecessary.